src/components/PeopleFilters.tsx

```diff
@@ -1,12 +1,107 @@
-export const PeopleFilters = () => {
+import classNames from 'classnames';
+import { useEffect } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+import { getSearchWith } from '../utils/searchHelper';
+import { SearchLink } from './SearchLink';
+
+type Props = {
+  peopleList: Person[];
+  setFilteredPeopleList: (list: Person[]) => void;
+};
+
+export const PeopleFilters: React.FC<Props> = ({
+  peopleList,
+  setFilteredPeopleList,
+}) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || null;
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sexList = ['All', 'Male', 'Female'];
+  const centuryList = ['16', '17', '18', '19', '20'];
+
+  const filterDependencies = async () => {
+    let filteredPeople = [...peopleList];
+
+    if (sex) {
+      filteredPeople = filteredPeople.filter(person => person.sex === sex);
+    }
+
+    if (query) {
+      const queryCheck = (name: string | null) => {
+        return name
+          ? name.toLowerCase().includes(query.toLowerCase())
+          : null;
+      };
+
+      filteredPeople = filteredPeople.filter(person => queryCheck(person.name)
+      || queryCheck(person.motherName)
+      || queryCheck(person.fatherName));
+    }
+
+    if (centuries.length) {
+      filteredPeople = filteredPeople.filter(person => {
+        const bornCentury = Math.ceil(person.born / 100).toString();
+
+        return centuries.includes(bornCentury);
+      });
+    }
+
+    setFilteredPeopleList(filteredPeople);
+  };
+
+  useEffect(() => {
+    filterDependencies();
+
+    return () => {
+      setFilteredPeopleList(peopleList);
+    };
+  }, [searchParams]);
+
+  const handleSexSelect = (gender: string) => {
+    switch (gender) {
+      case 'Female':
+        return 'f';
+
+      case 'Male':
+        return 'm';
+
+      case 'All':
+      default:
+        return null;
+    }
+  };
+
+  const handleQueryChange = async (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setSearchParams(
+      getSearchWith(
+        searchParams,
+        { query: event.target.value || null },
+      ),
+    );
```

Comment:
Find a value from the function call and then set it to the state, please. It's hard to read now

Reply:
Didn't catch it